scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -34,44 +34,6 @@
                  "compute cluster, unable to get node from node name" node-name)
       false)))
 
-(defn compute-num-waiting-synthetic-pods
-  "Given a compute cluster and a collection of pods, returns the number of
-  waiting synthetic pods in the cluster, or 0 if the cluster is not autoscaling"
-  [compute-cluster pods pool-name grace-period-seconds]
-  (if (cc/autoscaling? compute-cluster pool-name)
-    (let [synthetic-pods (filter controller/synthetic-pod->job-uuid pods)
-          waiting-synthetic-pods (filter (fn waiting-and-recent?
-                                           [^V1Pod pod]
-                                           (and
-                                             (or (-> pod
-                                                     .getStatus
-                                                     nil?)
-                                                 (-> pod
-                                                     api/pod->synthesized-pod-state
-                                                     :state
-                                                     (= :pod/waiting)))
-                                             ; We don't want to wait indefinitely because
-                                             ; synthetic pods didn't get scheduled. So, we
-                                             ; only count a synthetic pod as "recent" if it
-                                             ; was created in the last grace-period-seconds.
-                                             (-> pod
-                                                 .getMetadata
-                                                 .getCreationTimestamp
-                                                 (time/plus (time/seconds grace-period-seconds))
-                                                 (time/after? (time/now)))))
-                                         synthetic-pods)
-          waiting-synthetic-pods-in-pool (filter (fn matching-cook-pool-toleration?
-                                                   [^V1Pod pod]
-                                                   (let [pod-spec (.getSpec pod)
-                                                         tolerations (.getTolerations pod-spec)]
-                                                     (some (fn [^V1Toleration toleration]
-                                                             (and (= (.getKey toleration) "cook-pool")
-                                                                  (= (.getValue toleration) pool-name)))
-                                                           tolerations)))
-                                                 waiting-synthetic-pods)]
-      (count waiting-synthetic-pods-in-pool))
-    0))
-
```

Comment:
### Why?

We no longer need skip offer generation due to pending synthetic pods. Instead, we're using pod preemption and allowing job pods to preempt synthetic pods in order to avoid races for scheduling on new capacity between them. This function was only being used to determine whether or not skip offer generation.